prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirToHl7Converter.kt

```diff
@@ -205,7 +206,14 @@ class FhirToHl7Converter(
         element.hl7Spec.forEach { rawHl7Spec ->
             val resolvedHl7Spec = constantSubstitutor.replace(rawHl7Spec, context)
             try {
-                terser!!.set(resolvedHl7Spec, value)
+                val maybeTruncatedValue = context.translationFunctions?.truncateHL7Field(
```

Comment:
Only maybe? 😆

Reply:
It depends on if the field is configured to truncate or not. I used `maybe` because plenty of fields wont be truncated after this step and will just pass right through (also will occur if `translationFunctions` is missing entirely).